### ISSUE TITLE
View only feature

### ISSUE THREAD

**Issue by johnae** [any1/wayvnc#27]
Hey, thanks for this awesome program!

I wonder if there's any way to start wayvnc in "view only"-mode and if not whether that is something relatively easy to implement?

I'm not very knowledgeable when it comes to seat mgmt in sway as I never had much need for it but I did try disabling the virtual pointer/keyboard inputs, but that didn't seem to do anything.

**Comment by any1** [comment]
It is not currently a feature, but it would be very easy to implement. I recommend that you take a look. It's probably just a matter of optionally omitting the registration of some callbacks.

**Comment by johnae** [comment]
I managed to disable both keyboard and pointer by commenting out "lots of stuff". That's clearly not yet a feature but I'll see if I can make it into one.

On a different note - I've been using wayvnc at https://github.com/any1/wayvnc/commit/7ea17d04aa14aa117a0502b5dc9088a119698ffb which has been working fine. When I started trying to disable input though I went ahead on the tip of master but it seemed wayvnc wouldn't run properly after making my changes - naturally I figured I did something stupid... until I tried building and running it without my changes. It still wouldn't run... so I went back to the working commit and made my changes there and that would work.

I guess this might actually be a bug report outside of this issue...

**Comment by johnae** [comment]
And when I say it "wouldn't run" I actually means it starts and then doesn't output anything to the terminal (it used to output the FPS)... and I can't connect to it. Stracing it it seems to just hang on poll:

```sh
ioctl(4, DRM_IOCTL_I915_GEM_MMAP, 0x7ffdcfc0d4a0) = 0
ioctl(4, DRM_IOCTL_I915_GEM_BUSY, 0x7ffdcfc0d500) = 0
ioctl(4, DRM_IOCTL_I915_GEM_MADVISE, 0x7ffdcfc0d570) = 0
ioctl(4, DRM_IOCTL_I915_GEM_WAIT or DRM_IOCTL_RADEON_GEM_OP, 0x7ffdcfc0d580) = 0
futex(0x7fa90ef21608, FUTEX_WAKE_PRIVATE, 2147483647) = 0
prlimit64(0, RLIMIT_STACK, NULL, {rlim_cur=8192*1024, rlim_max=RLIM64_INFINITY}) = 0
mmap(NULL, 8392704, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_STACK, -1, 0) = 0x7fa904859000
mprotect(0x7fa90485a000, 8388608, PROT_READ|PROT_WRITE) = 0
clone(child_stack=0x7fa905058c30, flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, parent_tid=[21126], tls=0x7fa905059700, child_tidptr=0x7fa9050599d0) = 21126
prlimit64(0, RLIMIT_STACK, NULL, {rlim_cur=8192*1024, rlim_max=RLIM64_INFINITY}) = 0
mmap(NULL, 8392704, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_STACK, -1, 0) = 0x7fa904058000
mprotect(0x7fa904059000, 8388608, PROT_READ|PROT_WRITE) = 0
clone(child_stack=0x7fa904857c30, flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, parent_tid=[21127], tls=0x7fa904858700, child_tidptr=0x7fa9048589d0) = 21127
prlimit64(0, RLIMIT_STACK, NULL, {rlim_cur=8192*1024, rlim_max=RLIM64_INFINITY}) = 0
mmap(NULL, 8392704, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_STACK, -1, 0) = 0x7fa8f185a000
mprotect(0x7fa8f185b000, 8388608, PROT_READ|PROT_WRITE) = 0
clone(child_stack=0x7fa8f2059c30, flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, parent_tid=[21128], tls=0x7fa8f205a700, child_tidptr=0x7fa8f205a9d0) = 21128
prlimit64(0, RLIMIT_STACK, NULL, {rlim_cur=8192*1024, rlim_max=RLIM64_INFINITY}) = 0
mmap(NULL, 8392704, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_STACK, -1, 0) = 0x7fa8f1059000
mprotect(0x7fa8f105a000, 8388608, PROT_READ|PROT_WRITE) = 0
clone(child_stack=0x7fa8f1858c30, flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, parent_tid=[21129], tls=0x7fa8f1859700, child_tidptr=0x7fa8f18599d0) = 21129
futex(0x7ffdcfc0d7a0, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 0, NULL, FUTEX_BITSET_MATCH_ANY) = 0
futex(0x7fa90ef21230, FUTEX_WAKE_PRIVATE, 2147483647) = 0
futex(0x7fa90ef21228, FUTEX_WAKE_PRIVATE, 1) = 1
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
sendmsg(3, {msg_name=NULL, msg_namelen=0, msg_iov=[{iov_base="\17\0\0\0\1\0\10\0", iov_len=8}], msg_iovlen=1, msg_controllen=0, msg_flags=0}, MSG_DONTWAIT|MSG_NOSIGNAL) = 8
poll([{fd=6, events=POLLIN|POLLPRI}, {fd=3, events=POLLIN|POLLPRI}], 2, -1^C) = ? ERESTART_RESTARTBLOCK (Interrupted by signal)
```

**Comment by any1** [comment]
I have been doing some extensive changes to the event handling code. Did you compile with neatvnc and aml in subprojects? I also removed the FPS printout, so that's to be expected.

What happens in strace when a client tries to connect?

**Comment by johnae** [comment]
Nothing more is output from strace, client just hangs.

**Comment by johnae** [comment]
Moved the issues with tip of master to its own issue here: https://github.com/any1/wayvnc/issues/28

**Comment by any1** [comment]
Implemented by #126
